### PR TITLE
Add aten::_foreach_add.Scalar/ScalarList/List

### PR DESCRIPTION
**Pull request by fengyuan14**
1. Implement MultiTensorApply kernel template.
2. Add aten::_foreach_add.Scalar/ScalarList/List.

### CODE REVIEW

**Comment by fengyuan14** [comment]
image

**Comment by fengyuan14** [comment]
image

**Review by EikanWang — APPROVED** [review]
Nit: Please replace `dpcpp` with `xpu`

File: examples/test_foreach_list.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import torch
+import random
+from torch.testing._internal.common_utils import TestCase
+from torch.testing._internal.common_dtype import get_all_dtypes
+
+cpu_device = torch.device("cpu")
+dpcpp_device = torch.device("xpu")
```
Comment (fengyuan14) [inline examples/test_foreach_list.py:7]:
```suggestion
xpu_device = torch.device("xpu")
```

File: examples/test_foreach_list.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
+import torch
+import random
+from torch.testing._internal.common_utils import TestCase
+from torch.testing._internal.common_dtype import get_all_dtypes
+
+cpu_device = torch.device("cpu")
```
Comment (fengyuan14) [inline examples/test_foreach_list.py:6]:
```suggestion
```

File: examples/test_foreach_list.py (hunk 1 above)
Comment (fengyuan14) [inline examples/test_foreach_list.py:7]:
```suggestion
```

File: examples/test_foreach_scalar.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
+import torch
+import random
+from torch.testing._internal.common_utils import TestCase
+from torch.testing._internal.common_dtype import get_all_dtypes
+
+cpu_device = torch.device("cpu")
```
Comment (fengyuan14) [inline examples/test_foreach_scalar.py:6]:
```suggestion
```

File: examples/test_foreach_scalar.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+dpcpp_device = torch.device("xpu")
```
Comment (fengyuan14) [inline examples/test_foreach_scalar.py:7]:
```suggestion
```

File: examples/test_foreach_scalarlist.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
+import torch
+import random
+from torch.testing._internal.common_utils import TestCase
+from torch.testing._internal.common_dtype import get_all_dtypes
+
+cpu_device = torch.device("cpu")
```
Comment (fengyuan14) [inline examples/test_foreach_scalarlist.py:6]:
```suggestion
```

File: examples/test_foreach_scalarlist.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+dpcpp_device = torch.device("xpu")
```
Comment (fengyuan14) [inline examples/test_foreach_scalarlist.py:7]:
```suggestion
```

**Pull request merged**